### PR TITLE
Bug 1842865: Fix Reconcile loop return values

### PR DESCRIPTION
**Pull request by vimalk78**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1842865

### CODE REVIEW

File: pkg/controller/kibana/controller.go [hunk 1]
```diff
@@ -65,13 +66,18 @@ var (
 func (r *ReconcileKibana) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	es, err := k8shandler.GetElasticsearchCR(r.client, request.Namespace)
 	if err != nil {
-		return reconcileResult, fmt.Errorf("skipping kibana reconciliation in %q: %s", request.Namespace, err)
+		logrus.Infof("skipping kibana reconciliation in %q: %s", request.Namespace, err)
+		return reconcileResult, nil
```
Comment (periklis) [inline pkg/controller/kibana/controller.go:70]: Wouldn't inlining `reconcile.Result{RequeueAfter: reconcilePeriod}` make the case more readable?

File: pkg/controller/kibana/controller.go [hunk 2]
```diff
@@ -65,13 +66,18 @@ var (
...
 	}
 
 	esClient := elasticsearch.NewClient(es.Name, es.Namespace, r.client)
 	if err := kibana.Reconcile(request, r.client, esClient); err != nil {
-		return reconcileResult, err
+		if errors.IsNotFound(err) {
+			return reconcile.Result{Requeue: true}, nil
```
Comment (periklis) [inline pkg/controller/kibana/controller.go:76]: Why not reconciling after a specific period here?

Comment (vimalk78) [reply pkg/controller/kibana/controller.go:76]: i guess this reconciliation is immediate, earlier than 30 secs wait.

Comment (periklis) [reply pkg/controller/kibana/controller.go:76]: Ok if this the intention, I am fine.

File: pkg/controller/kibana/controller.go (hunk 1 above)
Comment (vimalk78) [reply pkg/controller/kibana/controller.go]: updated.

File: pkg/k8shandler/kibana/reconciler.go [hunk 3]
```diff
@@ -48,9 +48,6 @@ func Reconcile(request reconcile.Request, k8sClient client.Client, esClient elas
 
 	err := k8sClient.Get(context.TODO(), key, kibanaInstance)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return nil
-		}
```
Comment (ewolinetz) [inline pkg/k8shandler/kibana/reconciler.go:53]: why was this removed?
